starwars_api/models.py

```diff
@@ -55,27 +63,64 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.next_page = None
+        self.previous_page = None
+        self.results = []
+        self.current_record = 0
+        self.current_page = 0
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
-
+        while True:
+            if not self.results:
```

Comment:
ah okay. embarrassingly enough, I still keep getting confused about the cases that match `if not` so seeing the analogous `if` really helps, thanks. 

to confirm - `self.results` will never be `None` because Swapi will have raised the client error when it can't get the next page. yes?